File: scripts/scip/fixtures/go/service/service.go
package service

import (
	"example.com/taskapp/model"
	"strings"
)

type TaskService struct {
	tasks []*model.Task
}

func NewTaskService() *TaskService {
	return &TaskService{tasks: make([]*model.Task, 0)}
}

func (s *TaskService) AddTask(title string, priority model.Priority) *model.Task {
	task := model.NewTask(title, priority)
	s.tasks = append(s.tasks, task)
	return task
}

func (s *TaskService) CompleteTask(title string) bool {
	for _, task := range s.tasks {
		if task.Title == title {
			task.MarkDone()
			return true
		}
	}
	return false
}

func (s *TaskService) GetPending() []*model.Task {
	var pending []*model.Task
	for _, task := range s.tasks {
		if !task.Done {
			pending = append(pending, task)
		}
	}
	return pending
}

func (s *TaskService) Summary() string {
	var lines []string
	for _, task := range s.tasks {
		lines = append(lines, task.Display())
	}
	return strings.Join(lines, "\n")
}

File: scripts/scip/fixtures/dotnet/Service.cs
using System.Linq;
using System.Collections.Generic;

namespace TaskApp;

public class TaskService
{
    private readonly List<TaskItem> _tasks = new();

    public TaskItem AddTask(string title, Priority priority = Priority.Medium)
    {
        var task = new TaskItem(title, priority);
        _tasks.Add(task);
        return task;
    }

    public bool CompleteTask(string title)
    {
        var task = _tasks.FirstOrDefault(t => t.Title == title);
        if (task == null) return false;
        task.MarkDone();
        return true;
    }

    public List<TaskItem> GetPending()
    {
        return _tasks.Where(t => !t.Done).ToList();
    }

    public string Summary()
    {
        return string.Join("\n", _tasks.Select(t => t.Display()));
    }
}
